ETL/Python ETL/MatchHistoryETL.py

```diff
@@ -0,0 +1,266 @@
+# import os
+# os.getcwd()
+
+import constants
+import requests
+import pandas as pd
+import numpy as np
+import psycopg2
+import csv
+import datetime
+from psycopg2 import sql
+
+pd.set_option('display.max_rows', 100)
+pd.set_option('display.max_columns', 50)
+
+
+# In[2]:
+
+
+def calculate_matchweek(conn):
+    """
+        Finds out last registered matchweek for current season in Database. 
+        Returns last registerered matchweek plus one. If there's any problem 
+        with databse or season has not even started, then returns default matchweek "1"
+    """
+    try:
+        cursor = conn.cursor()
+        matchweek_query = f"SELECT MAX(\"MatchWeek\") FROM public.match_history WHERE \"Season\" = '{constants.CURRENT_SEASON_TAG}'"
+        cursor.execute(matchweek_query)
+        matchweek_result = cursor.fetchone()
+        if matchweek_result[0] is None:
+            return constants.DEFAULT_MATCHWEEK
+        else:
+            # Next matchweek
+            return matchweek_result[0] + 1
+    except psycopg2.Error as e:
+        print ("An error ocurred in database")
+        print (e)
+        return constants.DEFAULT_MATCHWEEK
+    finally:
+        cursor.close()
+
+
+# In[3]:
+
+
+def upsert_query(to_table, reference_column, dataframe):
+    """
+        Generates a Composed SQL object that contains all necessary SQL code to 
+        perform upserts into a postgresql table from dataframe as source.
+        If row attempted to insert is already present in database, then engine
+        will skip insertion. 
+
+        Input:
+            to_table - target table name in format 'schema.table_name' to generate inserts 
+            reference_column - Column name of your table that is considered PK to avoid overwritings
+            dataframe - pandas dataframe containing all rows to be inserted
+    """
+    date_config = sql.SQL('SET datestyle = "ISO, DMY"; ')
+    
+    insert_header = sql.SQL("INSERT INTO " + to_table + " ({fields})").format(
+        fields = sql.SQL(',').join([sql.Identifier(column) for column in dataframe.columns])
+    )
+    
+    values_array = []
+    for row in dataframe.iterrows():
+        values_array.append(sql.SQL(" ({values})").format(
+            values = sql.SQL(',').join([sql.Literal(value) for value in row[1].values])
+        ))
+    
+    insert_values = sql.SQL(' VALUES ') + sql.SQL(',').join(values_array)
+    
+    insert_condition = sql.SQL(" ON CONFLICT({conflict_col}) DO NOTHING").format(
+        conflict_col = sql.Identifier(reference_column)
+    )
+    
+    insert_statement = date_config + insert_header + insert_values + insert_condition
+   
+    return insert_statement
+
+
+# In[4]:
+
+
+# -- EXTRACTION
+
+website_frame = pd.read_csv(constants.CSV_SOURCE_URL)
+
+
+# In[6]:
+
+
+# -- TRANSFORMATION
+
+# Remove unwanted columns
+
+unwanted_cols = ["Div", "BWH", "BWD", "BWA", "IWH", "IWD", "IWA", "PSH", "PSD", "PSA", "WHH", "WHD", "WHA", "VCH", "VCD", "VCA",
```

Comment:
you can create a yaml for this.

Reply:
oh I did not know that yaml can be used for that too. I though it was only for configuration things. Could you give a short example of how would look like?